src/olympia/zadmin/models.py

```diff
@@ -149,6 +149,58 @@ def apply_validation(self, validation):
         self.notices = results['notices'] + compat['notices']
         self.valid = self.errors == 0
 
+        messages = results['messages']
+        message_ids = ['.'.join(msg['id']) for msg in messages]
+
+        message_summary = {}
+
+        for message, message_id in zip(messages, message_ids):
+            compat_type = message.get('compatibility_type')
+
+            if not compat_type:
+                compat_type = message.get('type')
+
+            if message_id not in message_summary:
+                message_summary[message_id] = (
+                    ValidationResultMessage.objects.get_or_create(
+                        message_id=message_id,
+                        validation_job=self.validation_job,
+                        defaults={
+                            'message': message.get('message', ''),
+                            'compat_type': compat_type,
+                        }
+                    )[0]
```

Comment:
That should be perfectly fine since it'd only happen if the `compat_type` changes by something in `amo-validator` which would imply a new validation job anyway so we don't have to update those.